test/suites/localzone/suite_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package localzone_test
+
+import (
+	"testing"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	corev1beta1 "sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/test"
+
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-provider-aws/test/pkg/environment/aws"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var env *aws.Environment
+var nodeClass *v1beta1.EC2NodeClass
+var nodePool *corev1beta1.NodePool
+
+func TestLocalZone(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		env = aws.NewEnvironment(t)
+	})
+	AfterSuite(func() {
+		env.Stop()
+	})
+	RunSpecs(t, "Chaos")
+}
+
+var _ = BeforeEach(func() {
+	env.BeforeEach()
+	nodeClass = env.DefaultEC2NodeClass()
+	// local zones don't support gp3 volumes, use gp2 instead
+	nodeClass.Spec.BlockDeviceMappings = append(nodeClass.Spec.BlockDeviceMappings, &v1beta1.BlockDeviceMapping{
+		DeviceName: lo.ToPtr("/dev/xvda"),
+		EBS: &v1beta1.BlockDevice{
+			VolumeSize: func() *resource.Quantity {
+				quantity, err := resource.ParseQuantity("80Gi")
+				Expect(err).To(BeNil())
+				return &quantity
+			}(),
+			VolumeType: lo.ToPtr("gp2"),
+			Encrypted:  lo.ToPtr(false),
+		},
+	})
+	nodePool = env.DefaultNodePool(nodeClass)
+	nodePool.Spec.Template.Spec.Requirements = append(nodePool.Spec.Template.Spec.Requirements, v1.NodeSelectorRequirement{
+		Key:      v1.LabelTopologyZone,
+		Operator: v1.NodeSelectorOpIn,
+		Values:   lo.Keys(lo.PickByValues(env.GetZones(), []string{"local-zone"})),
+	})
+})
+var _ = AfterEach(func() { env.Cleanup() })
+var _ = AfterEach(func() { env.AfterEach() })
+
+var _ = Describe("LocalZone", func() {
```

Comment:
Since this is in its own test suite, this will create another cluster. Is it possible for us to include this in another suite so that we can reduce the number of clusters we have to create? It seems unnecessary to create a whole cluster just to test a pod coming online.

Reply:
I agree that it seems excessive, the rationale was that this would be the only cluster with local zones enabled. Rather than modifying the existing suites to work around the limitations of local zones, or requiring each NodePool or NodeClass to filter out the zones (either by a topology selector or the subnet selector), this allows us to have a single cluster with local zone support.

Reply:
This makes sense to me since cluster creation is pretty cheap here and I expect our tests to scope down in this package when we have Kwok. This seems resonable to me given the trade-offs